cpp/farm_ng/core/logging/logger.h

```diff
@@ -40,19 +40,23 @@ std::string stringFromLogLevel(LogLevel level);
 // A logger that writes to std::cerr
 class StreamLogger {
  public:
-  static LogLevel const kDefaultLogLevel = LogLevel(LogLevel::info);
-
   // The header format is a {fmt}-style format string that may include the
-  //  named arguments {level}, {text}, {file}, {line}, {function}, {time},
-  //  {time_ms}.
+  //  named arguments {level}, {text}, {file}, {line}, {function}, {time}.
   void setHeaderFormat(std::string const& str);
   std::string getHeaderFormat() const;
 
   // Set the runtime log level
   void setLogLevel(LogLevel level);
-
   LogLevel getLogLevel();
 
+  // A type-erased generator of timestamp strings
+  struct LogClock {
+    std::function<std::string()> now;
+  };
```

Comment:
Any reason to not just do `using LogClock = std::function<std::string()>;` ?

Reply:
I thought about it, and liked this API better in terms of the type names and how it reads at the call-site. You might be right; I could go either way.